docs/fastddsgen/dataTypes/dataTypes.rst

```diff
@@ -183,6 +183,76 @@ In this case, the resulting C++ code will be:
    :start-after: // STRUCTURE_INHERITANCE
    :end-before: //!
 
+Optional members
+""""""""""""""""
+
+A member of a structure can be optional.
+This is achieved by writing the ``@optional`` annotation before the member.
+
+.. code-block:: omg-idl
+
+    struct StructWithOptionalMember
+    {
+        @optional octet octet_opt;
+    };
+
+An optional member is converted into a template class ``eprosima::fastcdr::optional<T>``, where ``T`` is the member's
+type.
+
+.. literalinclude:: /../code/FastDDSGenCodeTester.cpp
+   :language: c++
+   :start-after: // STRUCTURE_WITH_OPTIONAL
+   :end-before: //!
+
+Before reading the value of the optional member, it should be checked the optional contains a value using
+``has_value()`` function.
+Accessing a *null* optional throws a ``eprosima::fastcdr::exception::BadOptionalAccessException`` exception.
+
+.. literalinclude:: /../code/FastDDSGenCodeTester.cpp
+   :language: c++
+   :start-after: // ACCESSING_OPTIONAL_VALUE
+   :dedent: 4
+   :end-before: //!
+
+Extensibility
+"""""""""""""
+
+In order to support evolving types without breaking interoperability, the concept of type extensibility is supported by
+*Fast DDS-Gen*.
+There are three extensibility kinds: *final*, *appendable* and *mutable*.
+
+* *FINAL* extensibility indicates that the type is strictly defined. It is not possible to add members while maintaining
+  type assignability.
+* *APPENDABLE* extensibility indicates that two types, where one contains all of the members of the other plus
+  additional members appended to the end, may remain assignable.
+* *MUTABLE* extensibility indicates that two types may differ from one another in the additional, removal, and/or
+  transposition of members while remaining assignable.
+
+.. code-block:: omg-idl
+
+    @extensibility(FINAL)
+    struct FinalStruct
+    {
+        octet octet_opt;
+    };
+
+    @extensibility(APPENDABLE)
+    struct AppendableStruct
+    {
+        octet octet_opt;
+    };
+
+    @extensibility(MUTABLE)
+    struct MutableStruct
+    {
+        octet octet_opt;
+    };
+
+.. note::
+
+    XCDRv1 encoding algorithm is not able to manage correctly the deserialization of an appendable structure when it is
+    used as a member of another one.
+
```

Comment:
I suggest adding a note about the `@final`, `@appendable` and `@mutable` annotations. I know that they are also included in the table below but I think it would be helpful in case that the user only reads this section.

Reply:
I think is enough information the information in the table.